rustic-doc.el

```diff
@@ -72,6 +78,39 @@ Needs to be a function because of its reliance on
 (defvar helm-ag-success-exit-status)
 (declare-function helm-ag "ext:helm-ag")
 
+(defun rustic-doc--make-semver (&rest version-kv)
+  "Return a new semver-like version from VERSION-KV.
+VERSION-KV is a plist which must have :major key, and optionaly
+:minor and :patch. Default values for :minor and :patch are 0.
+Returned value is a list of the form (major minor patch)."
+  (let ((major (plist-get version-kv :major))
+        (minor (or (plist-get version-kv :minor) 0))
+        (patch (or (plist-get version-kv :patch) 0)))
+    (list major minor patch)))
+
+(defun rustic-doc--semver-from-string (version-str)
+  "Return a semver-like version from VERSION-STR."
+  (let* ((splitted-str (split-string version-str "\\."))
+         (str-length (length splitted-str)))
+    (mapcar #'string-to-number
+            ;; take only first 3 elements or less
+            (subseq splitted-str 0 (min 3 str-length)))))
```

Comment:
> `subseq`

This isn't in scope on my system and makes `defun` evaluation fail. This should be [seq-subseq](https://www.gnu.org/software/emacs/manual/html_node/elisp/Sequence-Functions.html#index-seq_002dsubseq), right?

Reply:
Interesting, by the time I made this PR, I had been using Doom Emacs, but my current configuration, that is made from scratch, doesn't have this function too. Maybe some of Doom Emacs dependencies added it to my environment. Added `seq-subseq` instead, tests seems working

Reply:
(as it happens, I'm using Doom currently, and this change was still required for it to work on my system)